internal/interpreter/ffi.rs

```diff
@@ -313,69 +268,57 @@ pub extern "C" fn slint_interpreter_struct_make_iter(stru: &StructOpaque) -> Str
     }
 }
 
-/// Get a property.
-/// The `out` parameter must be uninitialized. If this function returns true, the out will be initialized
-/// to the resulting value. If this function returns false, out is unchanged
+/// Get a property. Returns a null pointer if the property does not exist.
 #[no_mangle]
 pub unsafe extern "C" fn slint_interpreter_component_instance_get_property(
     inst: &ErasedItemTreeBox,
     name: Slice<u8>,
-    out: *mut ValueOpaque,
-) -> bool {
+) -> *mut Value {
     generativity::make_guard!(guard);
     let comp = inst.unerase(guard);
     match comp
         .description()
         .get_property(comp.borrow(), &normalize_identifier(std::str::from_utf8(&name).unwrap()))
     {
-        Ok(val) => {
-            std::ptr::write(out as *mut Value, val);
-            true
-        }
-        Err(_) => false,
+        Ok(val) => Box::into_raw(Box::new(val)),
+        Err(_) => std::ptr::null_mut(),
     }
 }
 
 #[no_mangle]
 pub extern "C" fn slint_interpreter_component_instance_set_property(
     inst: &ErasedItemTreeBox,
     name: Slice<u8>,
-    val: &ValueOpaque,
+    val: &Box<Value>,
 ) -> bool {
     generativity::make_guard!(guard);
     let comp = inst.unerase(guard);
     comp.description()
         .set_property(
             comp.borrow(),
             &normalize_identifier(std::str::from_utf8(&name).unwrap()),
-            val.as_value().clone(),
+            val.as_ref().clone(),
         )
         .is_ok()
 }
 
-/// Invoke a callback or function
-/// The `out` parameter must be uninitialized. If this function returns true, the out will be initialized
-/// to the resulting value. If this function returns false, out is unchanged
+/// Invoke a callback or function. Returns raw boxed value on success and null ptr on failure.
 #[no_mangle]
 pub unsafe extern "C" fn slint_interpreter_component_instance_invoke(
     inst: &ErasedItemTreeBox,
     name: Slice<u8>,
-    args: Slice<ValueOpaque>,
-    out: *mut ValueOpaque,
-) -> bool {
-    let args = std::mem::transmute::<Slice<ValueOpaque>, Slice<Value>>(args);
+    args: Slice<Box<Value>>,
+) -> *mut Value {
+    let args = args.iter().map(|vb| vb.as_ref().clone()).collect::<Vec<_>>();
```

Comment:
this is unfortunate that we need to clone everything, but i can't think of a better way.